zad2/warriors.rb

```diff
@@ -1,12 +1,181 @@
-class Warrior
-  def initialize(name:, level:)
+require 'faker'
+# Error that has to be raised if the wrong type of characters are entering the BattleArena (non-Character class)
+class ArenaEntryError < StandardError
+  def message
+    'You probably try to make some integers fight each other, you sadistic bastard! Check if the competitors are Characters before inviting them to the Arena.'
   end
+end
+
+# Basic character class
+class Character
+  attr_accessor :name
+  attr_accessor :level
+  MIN_LVL = 1
+  MAX_LVL = 99
 
+  # Constructor
+  def initialize name:, level:
+    @name = name #@name ?
+    @level = eval_level(level)
+  end
+  
+  # This method calculates the strength of characters attack
   def strength
+    self.level + [*1..12].sample
+  end
+  
+  # A character card: displays characters attributes.
+  def card
+    "#{@name} (lvl #{@level}), class: #{self.class}"
+  end
+
+
+  # Check if characters level has a valid value (belongs to [1:99]), if outside of the bounds: set the naerest valid value.
+  def eval_level level
+    @level = level.round()
```

Comment:
to `round` to w sumie trochę za dużo, jeśli chcemy się upewnić, że ktoś nie podał połówek, możemy użyć `level.to_i`, które utnie część po przecinku. Czasami lepiej nie robić feature'a z błędnego zachowania użytkownika :)

Reply:
Jak tak pomyślę o grach RPG to nawet ma sens, istotny jest ukończony poziom, nie najbliższy doświadczeniu - 72 i 87,34% to ciągle lvl 72. Przyjąłem :)